source/auth/auth.rst

```diff
@@ -863,13 +863,30 @@ mechanism_properties
 	AWS_SESSION_TOKEN
 		Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.
 
+
+Obtaining Credentials
+`````````````````````
 IAM Credentials
-   If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers
-   MUST use this secret key to generate a signature using HMAC-SHA256.
+   If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers MUST 
+   use this secret key to generate a signature using HMAC-SHA256. If a username is provided without a password (or vice-versa) drivers 
+   MUST raise an error. An example URI for authentication with MONGODB-IAM using IAM credentials is as follows:
+
+   ``mongodb://<access_key>:<secret_key>@mongodb.example.com/?authMechanism=MONGODB-IAM``
+
+Temporary IAM Credentials
+   If a username and password is not provided drivers MUST query the standard local AWS endpoint for temporary credentials. If 
```

Comment:
"If username and password are not"

Reply:
Done

Reply:
@ShaneHarvey please comment back or resolve.